libraries/cyclestreets-view/src/main/java/net/cyclestreets/views/CycleMapView.java

```diff
@@ -101,17 +106,28 @@ public Overlay overlayPushTop(final Overlay overlay) {
   public void onPause() {
     if (paused_)
       return;
+
+    // Some process seems to somehow set the co-ordinates to (85.051128, -180.0) and then
+    // pause/resume. Despite extensive debugging, I've not been able to figure out the cause.
+    // As a workaround, ignore ridiculous (ant)arctic co-ordinates - I can't imagine anyone needing
+    // cycle navigation in such places!
+    if (abs(getMapCenter().getLatitude()) > 80) {
+      Log.d(TAG, "Working around weird location setting bug");
+      return;
```

Comment:
This is the actual "fix".  Everything else is adding useful debugging lines, and minor refactors.